### PR TITLE
feat(ARCO-91): beef documentation

### PR DESCRIPTION
**Pull request by kuba-4chain**

### CODE REVIEW

File: doc/README.md
Comment (kuba-4chain) [inline doc/README.md:122]: I was hesitant to write this section. Let me know what you think, we can also remove it.

File: doc/README.md
Comment (arkadiuszos4chain) [inline doc/README.md:116]:
```suggestion
BEEF includes the transaction which constitutes the payment, as well as transactions whose outputs are used as inputs to the payment transaction (parent transactions) with their corresponding Merkle paths in the form of [BUMP](https://bsv.brc.dev/transactions/0074). In cases where the parent transaction is not yet mined, each ancestral transaction is included until the ancestor transaction is mined and has a corresponding Merkle path.
```

Comment (kuba-4chain) [reply doc/README.md]: Nice, done

**Review by arkadiuszos4chain — APPROVED** [review]

File: CHANGELOG.md
Comment (boecklim) [inline CHANGELOG.md:18]: 1.1.84 is not up to date anymore. Next version will be v1.1.86

File: doc/README.md
Comment (boecklim) [reply doc/README.md:122]: Looks good to me

File: doc/README.md
Comment (kuba-4chain) [reply doc/README.md]: Right, done

File: CHANGELOG.md
Comment (kuba-4chain) [reply CHANGELOG.md]: Changed to v1.1.87

**Review by boecklim — APPROVED** [review]

File: doc/README.md
Comment (sirdeggen) [inline doc/README.md:114]:
```suggestion
The second format is called Background Evaluation Extended Format, or BEEF, in short. BEEF was created to enable and facilitate Simplified Payment Verification (SPV) when sending transactions between peers which allows validation of transactions, the inputs of which may not yet be mined. Although this format is mainly used in peer-to-peer transactions, a peer ultimately has to submit the transaction to the nodes, and to help with that, Arc not only accepts that format, but also performs the SPV.
```

File: doc/README.md
Comment (sirdeggen) [inline doc/README.md:128]:
```suggestion
The Extended Format has been described in detail in [BRC-30](https://bsv.brc.dev/transactions/0030).
```

File: doc/README.md
Comment (sirdeggen) [inline doc/README.md:178]:
```suggestion
For this reason it is expected that transactions come in an extended format or BEEF. Transactions in standard format require a pretreatment to convert each tx into extended format to pass through the ARC pipeline.
```

File: doc/README.md
Comment (sirdeggen) [inline doc/README.md:180]:
```suggestion
That pretreatment detracts from the expected efficiency of the process because it requires extra requests to the Bitcoin network. Therefore, it is expected that the trend will be to use the EF or BEEF, which may be the only supported formats in the future.
```

**Review by sirdeggen — CHANGES_REQUESTED** [review]
approved other than my suggestions

**Comment by kuba-4chain** [comment]
Suggestions committed, waiting for your approval @sirdeggen :)

**Review by sirdeggen — APPROVED** [review]

**Pull request merged**
